scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala

```diff
@@ -431,9 +440,16 @@ class FormatWriter(formatOps: FormatOps) {
             !prevNonComment(formatToken).left
               .is[LeftParen] && // skip empty parentheses
             right.is[CloseDelim] && isNewline =>
-        sb.insert(
-          sb.length - left.syntax.length - prevFormatToken.between.length,
-          ",")
+        val indexOfComment = sb.lastIndexOf(left.syntax)
+        val index = sb.lastIndexOf(prevFormatToken.left.syntax, indexOfComment)
```

Comment:
Should we check the return values of `lastIndex` defensively?
I think we don't have to check the return values because we know that `sb.lastIndex` returns correct values here.